geopandas/geodataframe.py

```diff
@@ -41,14 +60,61 @@ class GeoDataFrame(GeoPandasBase, DataFrame):
     _metadata = ['crs', '_geometry_column_name']
 
     _geometry_column_name = DEFAULT_GEO_COLUMN_NAME
+    _pandas_cls = pd.Series
 
     def __init__(self, *args, **kwargs):
         crs = kwargs.pop('crs', None)
-        geometry = kwargs.pop('geometry', None)
-        super(GeoDataFrame, self).__init__(*args, **kwargs)
+
+        geometry = kwargs.pop('geometry', 'geometry')
+        [arg] = args
+
+        if isinstance(arg, BlockManager):
+            super(GeoDataFrame, self).__init__(arg, **kwargs)
+            self.crs = crs
+            self._geometry_column_name = geometry
+            return
+
+        gs = {}
+
+        if isinstance(arg, dict):
+            for i, (k, v) in list(enumerate(arg.items())):
+                if isinstance(v, GeoSeries):
+                    if 'columns' in kwargs:
+                        kwargs['columns'].pop(i)
+                    gs[k] = arg.pop(k)
+                    kwargs['index'] = v.index  # TODO: assumes consistent index
+
+        if isinstance(arg, (dict, list, pd.Series, np.ndarray)):
+            arg = pd.DataFrame(arg, **kwargs)
+
+        assert isinstance(arg, pd.DataFrame)
```

Comment:
do you typically leave such asserts? 
We could also raise a ValueError saying that the passed data is not supported